CoreScriptsRoot/Modules/BackpackScript.lua

```diff
@@ -1507,7 +1576,84 @@ local function OnVREnabled(prop)
 	if prop == "VREnabled" and UserInputService.VREnabled then
 		OnCoreGuiChanged(backpackType, StarterGui:GetCoreGuiEnabled(backpackType))
 		OnCoreGuiChanged(healthType, StarterGui:GetCoreGuiEnabled(healthType))
-		spawn(function() require(RobloxGui.Modules.BackpackScript3D) end)
+
+		local inventoryOpenStudSize = Vector2.new(11, 6.5)
+		local inventoryClosedStudSize = Vector2.new(11, 2)
+		local inventoryOpenPanelCF = CFrame.new(0, 4, 0) * CFrame.Angles(math.rad(-10), 0, 0)
+		local inventoryClosedPanelCF = CFrame.new(0, 2, 0) * CFrame.Angles(math.rad(-10), 0, 0)
+
+		local VRHub = require(RobloxGui.Modules.VR.VRHub)
+
+		BackpackScript.ModuleName = "Backpack"
+		BackpackScript.KeepVRTopbarOpen = true
+		BackpackScript.VRIsExclusive = true
+		BackpackScript.VRClosesNonExclusive = true
+		VRHub:RegisterModule(BackpackScript)
+
+		local BackpackPanel = Panel3D.Get(BackpackScript.ModuleName)
+		BackpackPanel:ResizeStuds(inventoryClosedStudSize.x, inventoryClosedStudSize.y)
+		BackpackPanel:SetType(Panel3D.Type.Fixed)
+		BackpackPanel:SetVisible(true)
+		MainFrame.Parent = BackpackPanel:GetGUI()
+
+
+		local panelLocalCF = inventoryClosedPanelCF
+		function BackpackPanel:PreUpdate(cameraCF, cameraRenderCF, userHeadCF, lookRay)
+			--the backpack panel needs to go in front of the user when they look at it.
+			--if they aren't looking, we should be updating self.localCF
+
+			local topbarPanel = Panel3D.Get("Topbar3D")
+			local panelOriginCF = topbarPanel.localCF or CFrame.new()
+			self.localCF = panelOriginCF * panelLocalCF
+		end
+
+		function BackpackPanel:OnUpdate()
+			-- Update transparency
+			for i = 1, HOTBAR_SLOTS do
+				local slot = Slots[i]
+				if slot then
+					slot:UpdateEquipView()
+				end
+			end
+			OpenInventoryButton.ImageTransparency = BackpackPanel.transparency
+			CloseInventoryButton.ImageTransparency = BackpackPanel.transparency
+			LeftBumperButton.ImageTransparency = BackpackPanel.transparency
+			RightBumperButton.ImageTransparency = BackpackPanel.transparency
+		end
+
+		OpenInventoryButton.Parent = MainFrame
+		CloseInventoryButton.Parent = InventoryFrame
+		-- LeftBumperButton.Parent = HotbarFrame
+		-- RightBumperButton.Parent = HotbarFrame
+
```

Comment:
LeftBumperButton and RightBumperButton are not parented. Are these going to be used in the future?